examples/ota-requestor-app/linux/main.cpp

```diff
@@ -189,5 +189,23 @@ int main(int argc, char * argv[])
 {
     VerifyOrDie(ChipLinuxAppInit(argc, argv, &cmdLineOptions) == 0);
     ChipLinuxAppMainLoop();
+
+    // If the event loop had been stopped due to an update being applied, boot into the new image
+    if (gRequestorCore.GetCurrentUpdateState() == OTARequestor::OTAUpdateStateEnum::kApplying)
+    {
+        if (kMaxFilePathSize <= strlen(kImageExecPath))
+        {
+            ChipLogError(SoftwareUpdate, "Buffer too small for the new image file path: %s", kImageExecPath);
+            return -1;
+        }
+
+        char execFilePathBuf[kMaxFilePathSize];
+        strncpy(execFilePathBuf, kImageExecPath, strlen(kImageExecPath));
+        argv[0] = execFilePathBuf;
+        execv(argv[0], argv);
```

Comment:
I guess calling this in `Apply()` caused the signal issues you mentioned on slack, right?